runtime/compiler/optimizer/J9TransformUtil.cpp

```diff
@@ -1006,184 +1005,232 @@ J9::TransformUtil::transformIndirectLoad(TR::Compilation *comp, TR::Node *node)
    return NULL;
    }
 
+bool
+J9::TransformUtil::foldReliableStaticFinalField(TR::Compilation *comp, TR::Node *node)
+   {
+   TR_ASSERT(node->getOpCode().isLoadVarDirect() && node->isLoadOfStaticFinalField(),
```

Comment:
I'm pretty sure that in sat least some cases on z static final loads can be represented as indirect loads. Doesn't this assert need to account for that?

Reply:
If they can still occur right now there needs to be a conditional return false rather than an assert until that deprecation is done.

Reply:
@andrewcraik Sure, I've updated the commit so that we return false instead of assert